File: backend/models/influx.py
from pydantic import BaseModel
from typing import List, Literal, Optional, Dict

class MeasurementList(BaseModel):
    measurements: List[str]


class Class(BaseModel):
    name: str
    measurements: List[str]

class QueryRequest(BaseModel):
    measurements: List[str]
    aggregation: Optional[Literal["mean", "max", "min", "sum"]] = "mean"
    start: str
    stop: str

class TimeseriesPoint(BaseModel):
    time: str
    value: float
    tags: Optional[Dict[str, str]] = None

class TimeseriesData(BaseModel):
    measurement: str
    points: List[TimeseriesPoint]

class ClassQueryData(BaseModel):
    class_name: str
    data: List[TimeseriesData]

    

File: frontend/src/types/influx.ts

export type MeasurementList = {
  measurements: string[];
};

export type Class = {
  name: string;
  measurements: string[];
};

export type TimeseriesPoint = {
  time: string; // ISO 8601
  value: number;
  tags?: Record<string, string>;
};

export type TimeseriesData = {
  measurement: string;
  points: TimeseriesPoint[];
};

export type ClassQueryData = {
  className: string;
  data: TimeseriesData[];
};

export type QueryRequest = {
  measurements: string[];
  aggregation?: "mean" | "max" | "min" | "sum";
  start: string;
  stop: string;
};
